Adding recipe for yggdrasil-python-rapidjson

langmm opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [X] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

langmm committed:
Adding recipe for yggdrasil-python-rapidjson

langmm committed:
Remove python minimum version constraint

langmm committed:
Get source distribution from PyPI & ensure that the local headers are used

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/yggdrasil-python-rapidjson/recipe.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/yggdrasil-python-rapidjson/recipe.yaml**:

 * ℹ️ PyPI default URL is now pypi.org, and not pypi.io. You may want to update the default source url.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/18136815901](https://github.com/conda-forge/conda-forge-webservices/actions/runs/18136815901). Examine the logs at this URL for more detail.</sub>

langmm committed:
Update pypi url & force use of Ninja on unix

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/yggdrasil-python-rapidjson/recipe.yaml```) and found it was in an excellent condition.

langmm committed:
Force Ninja generator via CMAKE_GENERATOR

langmm commented:
@conda-forge/help-python-c, ready for review!

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm committed:
Update recipes/yggdrasil-python-rapidjson/recipe.yaml

Remove numpy pinning following suggestion because the global version is higher than the pin

Co-authored-by: Daniel Nachun <daniel.nachun@gmail.com>

langmm committed:
Pin pandas to <3.0.0 until upstream package can be updated to support it

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm commented:
@danielnachun Thank you for taking the time to review this PR. Are there anything other changes I should make to allow this to move forward?

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm commented:
@conda-forge/help-python-c I made the requested changes, is there anything else I need to do for this to be merged?

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

ocefpaf commented on recipes/yggdrasil-python-rapidjson/recipe.yaml:
```diff
@@ -0,0 +1,82 @@
+  version: ${{ version }}
+
+source:
+  url: https://pypi.org/packages/source/${{ name[0] }}/${{ name }}/${{ python_name }}-${{ version }}.tar.gz
```
```suggestion
  url: https://pypi.org/packages/source/${{ name[0] }}/${{ name }}/yggdrasil_python_rapidjson-${{ version }}.tar.gz
```

ocefpaf commented on recipes/yggdrasil-python-rapidjson/recipe.yaml:
```diff
@@ -0,0 +1,82 @@
+context:
+  name: "yggdrasil-python-rapidjson"
+  python_name: "yggdrasil_python_rapidjson"
```
We are slowly dropping unnecessary variables. They can render the recipe quite hard to read and don't bring any advantages.

```suggestion
```

ocefpaf reviewed (commented):
Minor comment and it is good to go.

langmm committed:
Update recipes/yggdrasil-python-rapidjson/recipe.yaml

Remove use of python_name variable

Co-authored-by: Filipe <ocefpaf@gmail.com>

langmm committed:
Update recipes/yggdrasil-python-rapidjson/recipe.yaml

Remove use of python_name variable

Co-authored-by: Filipe <ocefpaf@gmail.com>

langmm committed:
Merge branch 'main' into yggdrasil-python-rapidjson

langmm commented:
@ocefpaf thank you for taking a look at this. With those changes are you able to approve the merge or does this still need approval from @danielnachun?

ocefpaf commented:
> @ocefpaf thank you for taking a look at this. With those changes are you able to approve the merge or does this still need approval from @danielnachun?

We usually let the original reviewer do the merge, but Daniel is super busy and I'm sure he'll won't mind me merging this one.
